scalafix/rules/src/main/scala/fix/Cats_v1_0_0.scala

```diff
@@ -29,160 +14,105 @@ case class RemoveCartesianBuilder(index: SemanticdbIndex)
 
   private[this] def wrapInParensIfNeeded(ctx: RuleCtx, t: Term): Patch = {
-    if (t.tokens.head.is[Token.LeftParen] && t.tokens.last
-        .is[Token.RightParen]) {
-      Patch.empty
-    } else {
-      ctx.addLeft(t.tokens.head, "(") + ctx.addRight(t.tokens.last, ")")
-    }
-  }
+    for {
+      head <- t.tokens.headOption
+      if !head.is[Token.LeftParen]
+      last <- t.tokens.lastOption
+      if !last.is[Token.RightParen]
+    } yield
+      ctx.addLeft(head, "(") +
+        ctx.addRight(last, ")")
+  }.asPatch
 
   override def fix(ctx: RuleCtx): Patch = {
     ctx.tree.collect {
-      case t: Term.ApplyInfix if t.op.matches(cartesianBuilders) =>
-        removeCartesianBuilderOp(ctx, t)
-      case t: Term.ApplyInfix if t.op.matches(cartesianOps) =>
-        wrapInParensIfNeeded(ctx, t.lhs)
-      case t: Term.Name => rename(ctx, t, renames)
+      case Term.ApplyInfix(_, cartesianBuilders(op: Term.Name), _, _) =>
+        replaceOpWithComma(ctx, op)
+      case Term.ApplyInfix(lhs, cartesianOps(_), _, _) =>
+        wrapInParensIfNeeded(ctx, lhs)
       case t @ q"import cats.syntax.cartesian._" =>
-        val usesPartialApplies = ctx.tree.collect {
-          case partialApplies(t: Term.Name) => ()
-        }.length > 0
+        val usesPartialApplies = ctx.tree.exists {
+          case partialApplies(_: Term.Name) => true
+          case _ => false
+        }
         if (usesPartialApplies) {
           ctx.addRight(t.tokens.last, "\n  import cats.syntax.apply._")
         } else {
           ctx.replaceTree(t, "import cats.syntax.apply._")
         }
-    }.asPatch
+    }.asPatch + ctx.replaceSymbols(renames.toSeq: _*)
   }
 }
 
 // ref: https://github.com/typelevel/cats/pull/1583
 case class RemoveUnapply(index: SemanticdbIndex)
     extends SemanticRule(index, "RemoveUnapply") {
 
-  private[this] val renames = Map(
-    Symbol("_root_.cats.Traverse.Ops.traverseU.") -> "traverse",
-    Symbol("_root_.cats.Foldable.Ops.traverseU_.") -> "traverse_",
-    Symbol("_root_.cats.Foldable.traverseU_.") -> "traverse_",
-    Symbol("_root_.cats.Traverse.Ops.sequenceU.") -> "sequence",
-    Symbol("_root_.cats.Foldable.Ops.sequenceU_.") -> "sequence_",
-    Symbol("_root_.cats.Foldable.sequenceU_.") -> "sequence_",
-    Symbol("_root_.cats.data.Func.appFuncU.") -> "appFunc",
-    Symbol("_root_.cats.free.FreeT.liftTU.") -> "liftT"
+  override def fix(ctx: RuleCtx): Patch = ctx.replaceSymbols(
+    "_root_.cats.Traverse.Ops.traverseU." -> "traverse",
+    "_root_.cats.Foldable.Ops.traverseU_." -> "traverse_",
+    "_root_.cats.Foldable.traverseU_." -> "traverse_",
+    "_root_.cats.Traverse.Ops.sequenceU." -> "sequence",
+    "_root_.cats.Foldable.Ops.sequenceU_." -> "sequence_",
+    "_root_.cats.Foldable.sequenceU_." -> "sequence_",
+    "_root_.cats.data.Func.appFuncU." -> "appFunc",
+    "_root_.cats.free.FreeT.liftTU." -> "liftT"
   )
-
-  private[this] def importeeName(importee: Importee): Option[Name] =
```

Comment:
This is btw handled by `index.Symbol`: https://github.com/scalacenter/scalafix/blob/bc4cc6cd0306f576a15d8e9f618f24d140da5c24/scalafix-core/shared/src/main/scala/scalafix/internal/util/EagerInMemorySemanticdbIndex.scala#L44

Reply:
nice!